R/pkg/inst/worker/daemon.R

```diff
@@ -31,7 +31,15 @@ inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
 while (TRUE) {
-  ready <- socketSelect(list(inputCon))
+  ready <- socketSelect(list(inputCon), timeout = 1)
+
+  # Terminate R workers in the parent process.
+  finishedChildren <- parallel:::selectChildren()
```

Comment:
This can be tested as below:

```bash
vi tmp.R
```

copy and paste

```R
p <- parallel:::mcfork()
if (inherits(p, "masterProcess")) {
  Sys.sleep(3L)
  parallel:::mcexit(0L)
}
print(parallel:::selectChildren())
Sys.sleep(4L)
print(parallel:::selectChildren())
```

This prints 

```bash
[1] TRUE
[1] 2505
```

For return values, please refer https://stat.ethz.ch/R-manual/R-devel/library/parallel/html/children.html.

For testing resource leak, this can be tested as below:

```R
for(i in 0:100000) {
  p <- parallel:::mcfork()
  if (inherits(p, "masterProcess")) {
    parallel:::mcexit(0L)
  }
  
  exitChildren <- parallel:::selectChildren()
  if (is.integer(exitChildren)) {
    lapply(exitChildren, function(x) { tools::pskill(x, tools::SIGUSR1) })
  }
}
```

The original code below:

```R
for(i in 0:100000) {
  p <- parallel:::mcfork()
  if (inherits(p, "masterProcess")) {
    tools::pskill(Sys.getpid(), tools::SIGUSR1)
    parallel:::mcexit(0L)
  }
}
```

will probably complain in `mcfork` due to the lack of resources (e.g., ran out of file descriptors).

Actual number can be checked via `watch -n 0.01 "lsof -c R | wc -l"` in another terminal.